packages/jellyfish-api-core/__tests__/category/account/futureswap.test.ts

```diff
@@ -0,0 +1,744 @@
+    symbol: 'TSLA',
+    fixedIntervalPriceId: 'TSLA/USD'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.setLoanToken({
+    symbol: 'AMZN',
+    fixedIntervalPriceId: 'AMZN/USD'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.setLoanToken({
+    symbol: 'DUSD',
+    fixedIntervalPriceId: 'DUSD/USD'
+  })
+  await testing.generate(1)
+
+  idTSLA = await testing.token.getTokenId('TSLA')
+  idDUSD = await testing.token.getTokenId('DUSD')
+
+  // create a vault and take loans
+  const vaultAddr = await testing.generateAddress()
+  const vaultId = await testing.rpc.loan.createVault({
+    ownerAddress: vaultAddr,
+    loanSchemeId: 'default'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.depositToVault({
+    vaultId: vaultId, from: collateralAddress, amount: '100000@DFI'
+  })
+
+  // wait till the price valid.
+  await testing.container.waitForPriceValid('TSLA/USD')
+
+  // take multiple loans
+  await testing.rpc.loan.takeLoan({
+    vaultId: vaultId,
+    to: collateralAddress,
+    amounts: ['300@TSLA', '500@DUSD', '100@AMZN']
+  })
+  await testing.generate(1)
+
+  // Futures setup
+  // set the dfip2203/active to false
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'false' } })
+  await testing.generate(1)
+
+  // set dfip2203 params
+  futInterval = 25
+  futRewardPercentage = 0.05
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/reward_pct': futRewardPercentage.toString(), 'v0/params/dfip2203/block_period': futInterval.toString() } })
+  await testing.generate(1)
+
+  // activat the dfip2203/active now
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'true' } })
+  await testing.generate(1)
+
+  // Retrive and verify gov vars
+  const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/active']).toStrictEqual('true')
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/reward_pct']).toStrictEqual(futRewardPercentage.toString())
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/block_period']).toStrictEqual(futInterval.toString())
+}
+
+async function getNextSettleBlock (): Promise<number> {
+  const blockCount = await testing.rpc.blockchain.getBlockCount()
+  const nextSettleBlock = blockCount + (futInterval - (blockCount % futInterval))
+  return nextSettleBlock
+}
+
+describe('futureSwap', () => {
+  beforeEach(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterEach(async () => {
+    await testing.container.stop()
+  })
+
+  it('should create futureswap dtoken to dusd', async () => {
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: '1@TSLA' })
+    await testing.generate(1)
+
+    const swapAmount = 1
```

Comment:
Apparently, swapAmount should be declared as a global variable in this file so you do not need to redeclare across many places.

Reply:
may be keep it within the test scope but will use as suggested in here -> https://github.com/DeFiCh/jellyfish/pull/1273#discussion_r843576320